{{cookiecutter.repostory_name}}/deploy.sh

```diff
@@ -1,7 +1,12 @@
 #!/bin/sh -eux
-docker-compose build
-docker-compose down -v
+PRIMARY_IMAGE_HASH=$(docker-compose build app | tee /dev/tty \
+    | grep "AS secondary-image" -B1 | head -1 | awk '{print $2}')
+docker-compose build  # build the rest of images
+docker-compose down -v --remove-orphans
 docker-compose up -d
 docker-compose exec app python manage.py migrate
 docker-compose exec app python manage.py collectstatic
-docker images --quiet --filter=dangling=true | xargs --no-run-if-empty docker rmi || true
+docker images --quiet --filter=dangling=true \
```

Comment:
won't it clean all the layers of primary image except the last one?

Reply:
No. After building images and running `docker images --quiet --filter=dangling=true` the only dangling image is the final hash of the first stage.

Reply:
That's odd. What if you'd modify requirements, would it then cause more images to be dangling? Will all of them be detected by the first run of `docker images --quiet --filter=dangling=true` or do we need to run it in a loop until it cleans up everything?

Reply:
I think that `--filter=dangling=true` doesn't include all images, just final and stages. I can list all the intermediate images with `docker images -a` but combining `docker images -a --filter=dangling=true` doesn't list intermediate images. On the other hand `docker rmi` removes an image and intermediate images that aren't used any more.

Reply:
ah ok now I get it